lib/Jsrt/ChakraCommon.h

```diff
@@ -721,6 +739,14 @@ typedef unsigned short uint16_t;
     /// </param>
     typedef void (CHAKRA_CALLBACK *JsFinalizeCallback)(_In_opt_ void *data);
 
+    /// <summary>
+    ///     A trace callback.
+    /// </summary>
+    /// <param name="data">
+    ///     The external data that was passed in when creating the object being traced.
+    /// </param>
+    typedef void (CHAKRA_CALLBACK *JsTraceCallback)(_In_opt_ void *data);
```

Comment:
Missed this guy - should be in chakracore.h

Reply:
I tried moving it to ChakraCore.h, but we still use this typedef in ChakraFull, so it seemed kind of annoying. I guess I can #ifdef out all the places though

Reply:
hmmmm, i thought the callback is only used with JsCreateTracedExternalObject and JsCreateCustomExternalObject which is in chakracore.h? Btw is there any reason these two API are not merged into one?

Reply:
I guess the APIs for these could be merged, but underneath they create very different types of objects so I'd rather hold off for now.

We share code for JsrtExternalObjects between ChakraCore and ChakraFull, and even though trace callback is only used in ChakraCore, it was referenced there in both builds... I moved it all behind ifdef now though

Reply:
SGTM, thx!